### PR TITLE
Add support for network filters on containers and label filters on volumes

### PR DESCRIPTION
**Pull request by wsong**

### CODE REVIEW

File: test/integration/api/ps.bats [hunk 1]
```diff
@@ -158,3 +158,27 @@ function teardown() {
...
+	start_docker_with_busybox 2
+	swarm_manage
+
+	docker_swarm network create -d bridge test1
+	docker_swarm network create -d bridge test2
+
+	sleep 5
+
+	docker_swarm run --name c1 -e constraint:node==node-0 --net node-0/test1 -d busybox:latest sleep 100
+	docker_swarm run --name c2 -e constraint:node==node-1 --net node-1/test1 -d busybox:latest sleep 100
```
Comment (wsong) [inline test/integration/api/ps.bats]: This line is consistently failing in the PR runs with "network node-1/test1 not found":

https://jenkins.dockerproject.org/job/Swarm-PRs%20(engine%201.13)/185/console

Any idea what is going on here? Shouldn't we be able to attach to newly-created networks right away?

Comment (wsong) [inline test/integration/api/ps.bats]: Ah, never mind, I just had to create the networks on each individual node.

@dongluochen and @nishanttotla, PTAL when you can.

**Comment by nishanttotla** [comment]
@wsong can you rebase this PR? the `engine-master` tests are failing due to an tests issue that was fixed recently.

File: test/integration/api/volume.bats [hunk 2]
```diff
@@ -117,7 +117,46 @@ function teardown() {
...
+	[ "$status" -eq 0 ]
+	# expect 3 lines: the header and one volume per node
+	[ "${#lines[@]}" -eq 3 ]
+	[[ "${lines[1]}" == *"testsubstrvol"* ]]
+	[[ "${lines[2]}" == *"testsubstrvol"* ]]
+
+	run docker_swarm volume ls --filter label=testlabel=notarealvalue
+	[ "$status" -eq 0 ]
+	# expect 3 lines: the header and one volume per node
+	[ "${#lines[@]}" -eq 1 ]
```
Comment (dongluochen) [inline test/integration/api/volume.bats]: Doesn't match `expect 3 lines`.

Comment (wsong) [inline test/integration/api/volume.bats]: Whoops, removed.

File: test/integration/api/volume.bats [hunk 3]
```diff
@@ -117,7 +117,46 @@ function teardown() {
...
+	[ "${#lines[@]}" -eq 3 ]
+	[[ "${lines[1]}" == *"testsubstrvol"* ]]
+	[[ "${lines[2]}" == *"testsubstrvol"* ]]
+
+	run docker_swarm volume ls --filter label=testlabel=notarealvalue
+	[ "$status" -eq 0 ]
+	# expect 1 line: just the header
+	[ "${#lines[@]}" -eq 1 ]
+	[[ "${lines[1]}" != *"testsubstrvol"* ]]
+	[[ "${lines[2]}" != *"testsubstrvol"* ]]
```
Comment (dongluochen) [inline test/integration/api/volume.bats]: If there is one line, you shouldn't use `lines[2]`.

Comment (wsong) [inline test/integration/api/volume.bats]: Fixed.

Comment (dongluochen) [inline test/integration/api/volume.bats]: I think if there is only header, that's `lines[0]`.

**Pull request merged**